src/main/java/com/influxdb/v3/client/internal/InfluxDBClientImpl.java

```diff
@@ -183,16 +185,48 @@ public Stream<Object[]> query(@Nonnull final String query,
         return queryData(query, parameters, options)
                 .flatMap(vector -> {
                     List<FieldVector> fieldVectors = vector.getFieldVectors();
-                    return IntStream
-                            .range(0, vector.getRowCount())
-                            .mapToObj(rowNumber -> {
-
-                                ArrayList<Object> row = new ArrayList<>();
-                                for (FieldVector fieldVector : fieldVectors) {
-                                    row.add(fieldVector.getObject(rowNumber));
-                                }
-                                return row.toArray();
-                            });
+                    var fields = vector.getSchema().getFields();
+                    return IntStream.range(0, vector.getRowCount())
+                           .mapToObj(rowNumber -> {
+                               ArrayList<Object> row = new ArrayList<>();
+                               for (int i = 0; i < fieldVectors.size(); i++) {
+                                   var schema = fields.get(i);
+                                   var metaType = schema.getMetadata().get("iox::column::type");
+                                   String valueType = metaType != null ? metaType.split("::")[2] : null;
+
+                                   Object value = fieldVectors.get(i).getObject(rowNumber);
+                                   if ("field".equals(valueType)) {
+                                       switch (metaType) {
+                                           case "iox::column_type::field::integer":
+                                           case "iox::column_type::field::uinteger":
+                                               var intValue = (Long) value;
+                                               row.add(intValue);
+                                               break;
+                                           case "iox::column_type::field::float":
+                                               var doubleValue = (Double) value;
+                                               row.add(doubleValue);
+                                               break;
+                                           case "iox::column_type::field::string":
+                                               var textValue = (Text) value;
+                                               row.add(textValue.toString());
+                                               break;
+                                           case "iox::column_type::field::boolean":
+                                               var boolValue = (Boolean) value;
+                                               row.add(boolValue);
+                                               break;
```

Comment:
There should be type checking to enhance the robustness of the client. For example, if `var intValue = (Long) value;` fails, the client should be able to recover from this situation—log the error and continue with the next field. Also, add tests for this scenario.

Reply:
@bednar 
So if var intValue = (Long) value fails, then we will set the value to null, right?

Reply:
Yes. We can enhance our type casting by something similar as is here:

- https://github.com/influxdata/influxdb-client-java/blob/94285a18c59beafb89bb3ac516f183722d7c202c/client-core/src/main/java/com/influxdb/query/internal/FluxResultMapper.java#L136.
- https://github.com/influxdata/influxdb-client-java/blob/94285a18c59beafb89bb3ac516f183722d7c202c/client-core/src/main/java/com/influxdb/query/internal/FluxResultMapper.java#L186